plugins/Channel/test.py

```diff
@@ -219,11 +225,88 @@ def join():
 
         self.assertRegexp('kban adlkfajsdlfkjsd', 'adlkfajsdlfkjsd is not in')
 
+    def testAccountKbanNoAccount(self):
+        self.irc.prefix = 'something!else@somehwere.else'
+        self.irc.nick = 'something'
+        self.irc.state.supported['ACCOUNTEXTBAN'] = 'a,account'
+        self.irc.state.supported['EXTBAN'] = '~,abc'
+        def join():
+            self.irc.feedMsg(ircmsgs.join(
+                self.channel, prefix='foobar!user@host.domain.tld'))
+        join()
+        self.irc.feedMsg(ircmsgs.op(self.channel, self.irc.nick))
+        self.assertKban('kban --account --exact foobar',
+                        'foobar!user@host.domain.tld')
+        join()
+        self.assertKban('kban --account foobar',
+                        'foobar!user@host.domain.tld')
```

Comment:
I wonder if it'd be better to throw an error if only `--account` is given and the user's not logged in. Exact match bans are pretty trivial to evade on most networks (even accidentally if your connection times out), to the point I doubt there's much value in the bot setting them in the first place.

Reply:
What if it uses the default banmask instead?

Reply:
I made it fall back to the default banmask, then to just the host (in case the default banmask is only `['account']` for some reason)